sound/soc/sof/pcm.c

```diff
@@ -356,9 +361,21 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 
 		/* fallthrough */
 	case SNDRV_PCM_TRIGGER_START:
+		/* No need to start it as the stream is already active */
```

Comment:
> is this legit?
> This state machine seems before convoluted since we don't support the INFO_RESUME flag.

Yes, exactly, as we don't support the INFO_RESUME flag, TRIGGER_RESUME actually won't happen, and replace, TRIGGER_START is sent at resume, that's why I need add check to the TRIGGER_START case also.

> So when would we actually get a TRIGGER_START while in D0i3?
> That sounds like copy/paste and not really thought through.

The trigger_start will be sent when resuming from S0ix(audio in D0i3), so this one here is needed and the trigger resume one can be actually removed(I keep both as once we change to support INFO_RESUME flag, it can still work).